src/event/writer.rs

```diff
@@ -161,9 +172,39 @@ impl EventWriter {
                 .expect("send error");
             rx_error
         } else {
+            self.event_handles.push_back(rx_flush);
             rx
         }
     }
+
+    /// Flush data.
+    ///
+    /// It will wait until all pending appends have acknowledgment.
+    ///
+    /// # Examples
+    /// ```ignore
+    /// let mut byte_writer = client_factory.create_byte_writer(segment).await;
+    /// let payload = vec![0; 8];
+    /// let size = event_writer.write_event(&payload).await;
+    /// event_writer.flush().await;
+    /// ```
+    pub async fn flush(&mut self) -> Result<(), Error> {
+        while self.event_handles.front().is_some() {
+            let mut receiver = self.event_handles.swap_remove_front(0).expect("get first handle");
+
+            let try_recv = receiver.try_recv();
+            match try_recv {
+                Err(TryRecvError::Closed) => {}
+                Err(TryRecvError::Empty) => {}
+                _ => {
+                    try_recv.map_err(|e| Error::InternalFailure {
+                        msg: format!("oneshot error {:?}", e),
+                    })??;
```

Comment:
is there an await missing here?
It looks like this method will just return without actually waiting on anything.
Also instead of `_` it's probably worth explicitly having an `Ok` case as this branch is currently handling both all unlisted errors and the success case.

Reply:
Issue here was that I wasnt sending anything to tx_flush sender. Now, I am so we wont have this issue.